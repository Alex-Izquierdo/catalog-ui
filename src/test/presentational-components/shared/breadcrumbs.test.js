import React from 'react';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store' ;
import { MemoryRouter, Route } from 'react-router-dom';
import { BreadcrumbItem } from '@patternfly/react-core';

import CatalogBreadrubms from '../../../presentational-components/shared/breadcrubms';

describe('<BreadCrumbs />', () => {
  let BreadcrumbWrapper;
  let mockStore;
  beforeEach(() => {
    BreadcrumbWrapper = ({ children, providerProps, ...props }) => (
      <MemoryRouter { ...props }>
        { children }
      </MemoryRouter>
    );
    mockStore = configureStore([]);
  });

  it('breadcrubms should render correctly empty breadcrumbs', () => {
    const store = mockStore({});
    const wrapper = mount(
      <BreadcrumbWrapper>
        <CatalogBreadrubms store={ store } />
      </BreadcrumbWrapper>
    );
    expect(wrapper.find(BreadcrumbItem)).toHaveLength(0);
  });

  it('breadcrubms should render correctly two portfolio children', () => {
    const store = mockStore({
      portfolioReducer: {
        selectedPortfolio: {
          name: 'Foo'
        }
      }
    });
    const wrapper = mount(
      <MemoryRouter initialEntries={ [ '/portfolios/detail/55' ] } >
        <Route path='/portfolios/detail/55' render={ props => <CatalogBreadrubms store={ store } { ...props } /> }/>
      </MemoryRouter>
    );

    const list = wrapper.find('ol');
    expect(list.children()).toHaveLength(2);
    expect(list.childAt(0).text()).toEqual('Portfolios');
    expect(list.childAt(1).text()).toEqual('Foo');
  });
});
