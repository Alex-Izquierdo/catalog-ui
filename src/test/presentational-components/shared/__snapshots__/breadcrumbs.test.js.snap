// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BreadCrumbs /> breadcrubms should render correctly empty breadcrumbs 1`] = `
<BreadcrumbWrapper>
  <MemoryRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "ln1uee",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "ln1uee",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Wrapper>
        <Provider
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
        >
          <withRouter(Connect(CatalogBreadrubms))>
            <Route>
              <Connect(CatalogBreadrubms)
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "key": "ln1uee",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "key": "ln1uee",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "key": "ln1uee",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <CatalogBreadrubms
                  dispatch={[Function]}
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "key": "ln1uee",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "key": "ln1uee",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "key": "ln1uee",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  reducers={Object {}}
                >
                  <Breadcrumb
                    aria-label="Breadcrumb"
                    className=""
                  >
                    <nav
                      aria-label="Breadcrumb"
                      className="pf-c-breadcrumb"
                    >
                      <ol
                        className="pf-c-breadcrumb__list"
                      />
                    </nav>
                  </Breadcrumb>
                </CatalogBreadrubms>
              </Connect(CatalogBreadrubms)>
            </Route>
          </withRouter(Connect(CatalogBreadrubms))>
        </Provider>
      </Wrapper>
    </Router>
  </MemoryRouter>
</BreadcrumbWrapper>
`;
